protocol/contracts/v2/TempleDebtToken.sol

```diff
@@ -170,30 +172,37 @@ contract TempleDebtToken is ITempleDebtToken, TempleElevatedAccess {
         if (_debtor == address(0)) revert CommonEventsAndErrors.InvalidAddress();
         if (_mintAmount == 0) revert CommonEventsAndErrors.ExpectedNonZero();
 
-        // Checkpoint the (base) debt and the (risk premium) debt for this borrower
-        uint256 _totalPrincipalAndBase = _compoundedBaseInterest();
-        _checkpointBase(_totalPrincipalAndBase);
-        uint256 _totalBaseShares = baseShares;
-
+        // First checkpoint both the base interest and (risk premium) interest of the debtor.
+        // Use RO (read only) to delay the updating of the baseCache storage to the end since it needs updating later anyway.
+        BaseCache memory _baseCache = _getBaseCacheRO();
         Debtor storage debtor = debtors[_debtor];
-        _checkpointDebtor(debtor);              
-        
+
+        // Get the debtor cache and update it's storage upfront since the debtor interest doesn't change later
+        DebtorCache memory _debtorCache = _getDebtorCache(_baseCache, debtor);
+
         // Calculate the number of (base) debt shares for this borrow amount. Ensure this is rounded down (same as EIP-4626)
-        uint256 sharesAmount = _debtToShares(_mintAmount, _totalPrincipalAndBase, _totalBaseShares, false);
+        uint256 newSharesAmount = _debtToShares(
+            _mintAmount, 
+            _baseCache.totalPrincipalAndBaseInterest, 
+            _baseCache.baseShares, 
+            false
+        );
 
         // Update the contract state
         {
             // Add the shares to the debtor and total
-            debtor.baseShares += sharesAmount.encodeUInt128();
-            baseShares = _totalBaseShares + sharesAmount;
+            debtor.baseShares = _debtorCache.baseShares + newSharesAmount.encodeUInt128();
+            baseShares = _baseCache.baseShares + newSharesAmount;
 
             // The principal borrowed now increases (which affects the risk premium interest accrual)
             // and also the (base rate) checkpoint representing the principal+base interest
-            debtor.principal += _mintAmount.encodeUInt128();
+            debtor.principal = _debtorCache.principal = _debtorCache.principal + _mintAmount.encodeUInt128();
             totalPrincipal += _mintAmount;
-            baseCheckpoint += _mintAmount;
+            baseCheckpoint = _baseCache.totalPrincipalAndBaseInterest + _mintAmount;
+            baseCheckpointTime = uint32(block.timestamp);
         }
 
+        emit DebtorBalance(_debtor, _debtorCache.principal, _debtorCache.baseInterest, _debtorCache.riskPremiumInterest);
```

Comment:
New event added as it is very useful for subgraph to know the updated splits of principal/interest when debt is minted/burned